core/frontend/src/tile/map/WmsUtilities.ts

```diff
@@ -13,4 +16,28 @@ export class WmsUtilities {
     const lastIndex = url.lastIndexOf("?");
     return lastIndex > 0 ? url.slice(0, lastIndex) : url;
   }
+
+  /**
+ * fetch XML from HTTP request
+ * @param url server URL to address the request
+ * @internal
+ */
+  public static async fetchXml(url: string, credentials?: RequestBasicCredentials): Promise<string> {
+
+    let headers: Headers|undefined;
+    if (credentials && credentials.user && credentials.password) {
+      headers = new Headers();
+      setBasicAuthorization(headers, credentials);
+    }
+
+    let response = await fetch(url, { method: "GET", headers });
+    if (!credentials && response.status === 401 && headersIncludeAuthMethod(response.headers, ["ntlm", "negotiate"])) {
+    // We got a http 401 challenge, lets try SSO (i.e. Windows Authentication)
+      response = await fetch(url, { method: "GET", credentials: "include" });
+    }
+
+    if (response.status !== 200)
+      throw new HttpResponseError(response.status, await response.text());
+    return response.text();
```

Comment:
is this supposed to be awaited?

Reply:
`WmsUtilities.fetchXml` already returns a Promise<string>, so does it matter to await inside fetchXml if anyway the caller of `WmsUtilities.fetchXml`  is expected to await?